napari/layers/shapes/shapes.py

```diff
@@ -1624,9 +1584,40 @@ def mode(self):
         """
         return str(self._mode)
 
+    @staticmethod
+    def _mode_help(mode):
+        from . import _shapes_key_bindings as kb
+
+        help_li = []
+        shortcuts = get_settings().shortcuts.shortcuts
+
+        for func, mode_ in (
+            (kb.activate_add_rectangle_mode, Mode.ADD_RECTANGLE),
+            (kb.activate_add_ellipse_mode, Mode.ADD_ELLIPSE),
+            (kb.activate_add_line_mode, Mode.ADD_LINE),
+            (kb.activate_add_path_mode, Mode.ADD_PATH),
+            (kb.activate_add_polygon_mode, Mode.ADD_POLYGON),
+            (kb.activate_direct_mode, Mode.DIRECT),
+            (kb.activate_select_mode, Mode.SELECT),
+            (kb.activate_shape_pan_zoom_mode, Mode.PAN_ZOOM),
+            (kb.activate_vertex_insert_mode, Mode.VERTEX_INSERT),
+            (kb.activate_vertex_remove_mode, Mode.VERTEX_REMOVE),
```

Comment:
I think that only a subset of this actions should be selected

Reply:
Which subset and why?

Reply:
(Also, rather than get stuck in a long discussion about this, I suggest removing Shapes layer support in this PR and then adding it in an followup PR.)

Reply:
The produces string is very long an maybe it is worth to not inform about every shortcuts for Shapes layer.